### ISSUE TITLE
issue with syncing on gmail account

### ISSUE THREAD

**Issue by euglevi** [neomutt/neomutt#3519]
I have a weird issue when syncing the mailbox in my gmail account. When I read a new email, sync the mailbox, quit and get back in again, the email is correctly shown as read. When I add to these steps deleting a few emails before or after reading the new email, then the read email is shown up as new again on entering again neomutt. What could it be?

If needed, I can provide my configuration files and the neomutt log.

**Comment by gahr** [comment]
is this a local maildir sync'd from IMAP using something like mbsync?

**Comment by brauner** [comment]
No, it's just using the built-in imap capabilities of neomutt. The config is mostly:

```
## Receive options.
set imap_user=foo@bar.baz
set folder=imaps://imap.gmail.com
set imap_pass=$my_pw
set spoolfile=imaps://imap.gmail.com/INBOX
set record="imaps://imap.gmail.com/[Gmail]/Sent Mail"
set postponed="imaps://imap.gmail.com/[Gmail]/Drafts"
set trash='+[Gmail]/Trash'

## Send options.
set realname='My Name'
set from='my@mail.com'
set use_from=yes
set smtp_url     = "smtp://brauner@mail.kernel.org:587/"
set smtp_pass    = $my_other_pass
set ssl_starttls = yes
```

**Comment by brauner** [comment]
As far as I can tell it started when I updated my trash settings `set trash='+[Gmail]/Trash'`

**Comment by euglevi** [comment]
I moved to using a combination of lieer and notmuch (which works very well), but I was not using mbsync. I was using IMAP with [oauth2 authentication](https://github.com/neomutt/neomutt/blob/main/contrib/oauth2/mutt_oauth2.py.README).  I had the same folders as @brauner.

**Comment by gahr** [comment]
Interesting.. I cannot reproduce this. So you say

* open a GMail mailbox with new mail
* delete an (old) email
* read the new mail
* quit
* reopen
* -> the email you read is still marked as new?

**Comment by gahr** [comment]
now I can repro it with some minimal config

**Comment by gahr** [comment]
Indeed, I can see that in the bad case (delete + read), we're not sending a `UID STORE <uid> +FLAGS.SILENT (\Seen)`

**Comment by brauner** [comment]
Did this ever get fixed? I'm still having issues with NeoMutt 20240425

**Comment by gahr** [comment]
Hi @brauner , no, I fear nobody's looked at it and *I* forgot. Do you want to try to take a look? I don't think I'll be able to get to it soon(ish).

**Comment by brauner** [comment]
> Hi [@brauner](https://github.com/brauner) , no, I fear nobody's looked at it and _I_ forgot. Do you want to try to take a look? I don't think I'll be able to get to it soon(ish).

Ok, I tried my best.
